src/style/table.rs

```diff
@@ -68,3 +66,33 @@ pub enum TableComponent {
     BottomLeftCorner,
     BottomRightCorner,
 }
+
+impl TableComponent {
+    const fn components() -> [TableComponent; 19] {
+        [
+            TableComponent::LeftBorder,
+            TableComponent::RightBorder,
+            TableComponent::TopBorder,
+            TableComponent::BottomBorder,
+            TableComponent::LeftHeaderIntersection,
+            TableComponent::HeaderLines,
+            TableComponent::MiddleHeaderIntersections,
+            TableComponent::RightHeaderIntersection,
+            TableComponent::VerticalLines,
+            TableComponent::HorizontalLines,
+            TableComponent::MiddleIntersections,
+            TableComponent::LeftBorderIntersections,
+            TableComponent::RightBorderIntersections,
+            TableComponent::TopBorderIntersections,
+            TableComponent::BottomBorderIntersections,
+            TableComponent::TopLeftCorner,
+            TableComponent::TopRightCorner,
+            TableComponent::BottomLeftCorner,
+            TableComponent::BottomRightCorner,
+        ]
+    }
```

Comment:
> Manually implemented iterators for enums are usually a bad idea as it's really easy to forget to adjust them if new variants are added. And there's no way to throw a lint error for stuff like this.

Yes. This is the original reason we use a macro to generate.

I try to use an internal method return `[TableComponent; 19]` so that the compiler can help a bit.